app/controllers/application_controller.rb

```diff
@@ -3,8 +3,114 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+    enable :sessions
+    set :session_secret, "secret"
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :'index'
+  end
+
+  get '/signup' do
+    Helpers.is_logged_in?(session) ? (redirect "/tweets") : (erb :'signup')
+  end
+
+  post '/signup' do
+    @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    if @user.save
+      session[:id] = @user.id
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/tweets' do
+    if Helpers.is_logged_in?(session)
+      @user = Helpers.current_user(session)
+      erb :'show'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    @tweet = Tweet.new(content: params[:content], user_id: session[:id])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.